dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/handle/HandleUtilTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.handle;
+
+import static com.google.edwmigration.dumper.application.dumper.handle.HandleUtil.createConfig;
+import static com.google.edwmigration.dumper.application.dumper.handle.HandleUtil.templateFromDataSource;
+import static com.google.edwmigration.dumper.application.dumper.handle.HandleUtil.verifyJdbcConnection;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.zaxxer.hikari.HikariConfig;
+import java.sql.SQLException;
+import javax.sql.DataSource;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HandleUtilTest {
+
+  @Test
+  public void templateFromDataSource_success_fetchSizeMatches() {
+    DataSource dataSource = mock(DataSource.class);
+
+    JdbcTemplate template = templateFromDataSource(dataSource);
+
+    assertEquals(1024, template.getFetchSize());
```

Comment:
it is not so obvious, why `fetchSize` should be equal to `1024`. And it is set to 1024 inside code, not even sure, if it is worth to have test for that.

Reply:
This was the value before. Unfortunately it's not documented why 1024 was chosen.

The intent of this test is to verify that the developer didn't forget to set the custom fetch size e.g. by creating a `JdbcTemplate` directly, without going through the util method.

This verification could be done with a named value or using mocks, but those options would complicate the test code.